### ISSUE TITLE
Install does not work with node 12.x.x

### ISSUE THREAD

**Issue by pubkey** [bitchan/eccrypto#42]
Installing eccrypto on nodejs version 12 will break on the build.

```bash
node -v
> v12.7.0
```

It can be reproduced by running `npm install eccrypto` in an empty project.

Error output:

```bash

make: Entering directory '/home/dnl/workspace/test/node_modules/secp256k1/build'
  CXX(target) Release/obj.target/secp256k1/src/addon.o
In file included from ../../nan/nan_converters.h:67:0,
                 from ../../nan/nan.h:221,
                 from ../src/addon.cc:2:
../../nan/nan_converters_43_inl.h: In static member function ‘static Nan::imp::ToFactoryBase<v8::Boolean>::return_t Nan::imp::ToFactory<v8::Boolean>::convert(v8::Local<v8::Value>)’:
../../nan/nan_converters_43_inl.h:18:51: warning: ‘v8::MaybeLocal<v8::Boolean> v8::Value::ToBoolean(v8::Local<v8::Context>) const’ is deprecated: ToBoolean can never throw. Use Local version. [-Wdeprecated-declarations]
       val->To ## TYPE(isolate->GetCurrentContext())                            \
                                                   ^
../../nan/nan_converters_43_inl.h:22:1: note: in expansion of macro ‘X’
 X(Boolean)
 ^
In file included from /home/dnl/.node-gyp/12.7.0/include/node/v8-internal.h:14:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/v8.h:25,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node.h:63,
                 from ../src/addon.cc:1:
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:2531:59: note: declared here
                 V8_WARN_UNUSED_RESULT MaybeLocal<Boolean> ToBoolean(
                                                           ^
/home/dnl/.node-gyp/12.7.0/include/node/v8config.h:311:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../../nan/nan_converters.h:67:0,
                 from ../../nan/nan.h:221,
                 from ../src/addon.cc:2:
../../nan/nan_converters_43_inl.h: In static member function ‘static Nan::imp::ValueFactoryBase<bool>::return_t Nan::imp::ToFactory<bool>::convert(v8::Local<v8::Value>)’:
../../nan/nan_converters_43_inl.h:37:57: warning: ‘v8::Maybe<bool> v8::Value::BooleanValue(v8::Local<v8::Context>) const’ is deprecated: BooleanValue can never throw. Use Isolate version. [-Wdeprecated-declarations]
   return val->NAME ## Value(isolate->GetCurrentContext());                     \
                                                         ^
../../nan/nan_converters_43_inl.h:40:1: note: in expansion of macro ‘X’
 X(bool, Boolean)
 ^
In file included from /home/dnl/.node-gyp/12.7.0/include/node/v8-internal.h:14:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/v8.h:25,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node.h:63,
                 from ../src/addon.cc:1:
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:2569:51: note: declared here
                 V8_WARN_UNUSED_RESULT Maybe<bool> BooleanValue(
                                                   ^
/home/dnl/.node-gyp/12.7.0/include/node/v8config.h:311:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../../nan/nan_new.h:189:0,
                 from ../../nan/nan.h:222,
                 from ../src/addon.cc:2:
../../nan/nan_implementation_12_inl.h: In static member function ‘static Nan::imp::FactoryBase<v8::Function>::return_t Nan::imp::Factory<v8::Function>::New(Nan::FunctionCallback, v8::Local<v8::Value>)’:
../../nan/nan_implementation_12_inl.h:105:32: error: no matching function for call to ‘v8::Function::New(v8::Isolate*&, void (&)(const v8::FunctionCallbackInfo<v8::Value>&), v8::Local<v8::Object>&)’
                           , obj));
                                ^
In file included from /home/dnl/.node-gyp/12.7.0/include/node/node.h:63:0,
                 from ../src/addon.cc:1:
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:4132:31: note: candidate: static v8::MaybeLocal<v8::Function> v8::Function::New(v8::Local<v8::Context>, v8::FunctionCallback, v8::Local<v8::Value>, int, v8::ConstructorBehavior, v8::SideEffectType)
   static MaybeLocal<Function> New(
                               ^~~
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:4132:31: note:   no known conversion for argument 1 from ‘v8::Isolate*’ to ‘v8::Local<v8::Context>’
In file included from ../src/addon.cc:2:0:
../../nan/nan.h: In constructor ‘Nan::Utf8String::Utf8String(v8::Local<v8::Value>)’:
../../nan/nan.h:1064:78: warning: ‘v8::Local<v8::String> v8::Value::ToString(v8::Isolate*) const’ is deprecated: Use maybe version [-Wdeprecated-declarations]
       v8::Local<v8::String> string = from->ToString(v8::Isolate::GetCurrent());
                                                                              ^
In file included from /home/dnl/.node-gyp/12.7.0/include/node/v8-internal.h:14:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/v8.h:25,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node.h:63,
                 from ../src/addon.cc:1:
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:2551:31: note: declared here
                 Local<String> ToString(Isolate* isolate) const);
                               ^
/home/dnl/.node-gyp/12.7.0/include/node/v8config.h:311:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../src/addon.cc:2:0:
../../nan/nan.h: In member function ‘void Nan::AsyncWorker::SaveToPersistent(const char*, const v8::Local<v8::Value>&)’:
../../nan/nan.h:1855:64: warning: ‘bool v8::Object::Set(v8::Local<v8::Value>, v8::Local<v8::Value>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     New(persistentHandle)->Set(New(key).ToLocalChecked(), value);
                                                                ^
In file included from /home/dnl/.node-gyp/12.7.0/include/node/v8-internal.h:14:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/v8.h:25,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node.h:63,
                 from ../src/addon.cc:1:
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:3366:26: note: declared here
                     bool Set(Local<Value> key, Local<Value> value));
                          ^
/home/dnl/.node-gyp/12.7.0/include/node/v8config.h:326:3: note: in definition of macro ‘V8_DEPRECATE_SOON’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../src/addon.cc:2:0:
../../nan/nan.h: In member function ‘void Nan::AsyncWorker::SaveToPersistent(const v8::Local<v8::String>&, const v8::Local<v8::Value>&)’:
../../nan/nan.h:1861:42: warning: ‘bool v8::Object::Set(v8::Local<v8::Value>, v8::Local<v8::Value>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     New(persistentHandle)->Set(key, value);
                                          ^
In file included from /home/dnl/.node-gyp/12.7.0/include/node/v8-internal.h:14:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/v8.h:25,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node.h:63,
                 from ../src/addon.cc:1:
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:3366:26: note: declared here
                     bool Set(Local<Value> key, Local<Value> value));
                          ^
/home/dnl/.node-gyp/12.7.0/include/node/v8config.h:326:3: note: in definition of macro ‘V8_DEPRECATE_SOON’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../src/addon.cc:2:0:
../../nan/nan.h: In member function ‘void Nan::AsyncWorker::SaveToPersistent(uint32_t, const v8::Local<v8::Value>&)’:
../../nan/nan.h:1867:44: warning: ‘bool v8::Object::Set(uint32_t, v8::Local<v8::Value>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     New(persistentHandle)->Set(index, value);
                                            ^
In file included from /home/dnl/.node-gyp/12.7.0/include/node/v8-internal.h:14:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/v8.h:25,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node.h:63,
                 from ../src/addon.cc:1:
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:3375:26: note: declared here
                     bool Set(uint32_t index, Local<Value> value));
                          ^
/home/dnl/.node-gyp/12.7.0/include/node/v8config.h:326:3: note: in definition of macro ‘V8_DEPRECATE_SOON’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../src/addon.cc:2:0:
../../nan/nan.h: In member function ‘v8::Local<v8::Value> Nan::AsyncWorker::GetFromPersistent(const char*) const’:
../../nan/nan.h:1873:61: warning: ‘v8::Local<v8::Value> v8::Object::Get(v8::Local<v8::Value>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
         New(persistentHandle)->Get(New(key).ToLocalChecked()));
                                                             ^
In file included from /home/dnl/.node-gyp/12.7.0/include/node/v8-internal.h:14:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/v8.h:25,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node.h:63,
                 from ../src/addon.cc:1:
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:3419:55: note: declared here
   V8_DEPRECATE_SOON("Use maybe version", Local<Value> Get(Local<Value> key));
                                                       ^
/home/dnl/.node-gyp/12.7.0/include/node/v8config.h:326:3: note: in definition of macro ‘V8_DEPRECATE_SOON’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../src/addon.cc:2:0:
../../nan/nan.h: In member function ‘v8::Local<v8::Value> Nan::AsyncWorker::GetFromPersistent(const v8::Local<v8::String>&) const’:
../../nan/nan.h:1879:55: warning: ‘v8::Local<v8::Value> v8::Object::Get(v8::Local<v8::Value>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     return scope.Escape(New(persistentHandle)->Get(key));
                                                       ^
In file included from /home/dnl/.node-gyp/12.7.0/include/node/v8-internal.h:14:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/v8.h:25,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node.h:63,
                 from ../src/addon.cc:1:
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:3419:55: note: declared here
   V8_DEPRECATE_SOON("Use maybe version", Local<Value> Get(Local<Value> key));
                                                       ^
/home/dnl/.node-gyp/12.7.0/include/node/v8config.h:326:3: note: in definition of macro ‘V8_DEPRECATE_SOON’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../src/addon.cc:2:0:
../../nan/nan.h: In member function ‘v8::Local<v8::Value> Nan::AsyncWorker::GetFromPersistent(uint32_t) const’:
../../nan/nan.h:1884:57: warning: ‘v8::Local<v8::Value> v8::Object::Get(uint32_t)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     return scope.Escape(New(persistentHandle)->Get(index));
                                                         ^
In file included from /home/dnl/.node-gyp/12.7.0/include/node/v8-internal.h:14:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/v8.h:25,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node.h:63,
                 from ../src/addon.cc:1:
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:3423:55: note: declared here
   V8_DEPRECATE_SOON("Use maybe version", Local<Value> Get(uint32_t index));
                                                       ^
/home/dnl/.node-gyp/12.7.0/include/node/v8config.h:326:3: note: in definition of macro ‘V8_DEPRECATE_SOON’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from /usr/include/c++/7/cassert:44:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node_object_wrap.h:26,
                 from ../../nan/nan.h:54,
                 from ../src/addon.cc:2:
../../nan/nan_object_wrap.h: In destructor ‘virtual Nan::ObjectWrap::~ObjectWrap()’:
../../nan/nan_object_wrap.h:24:25: error: ‘class Nan::Persistent<v8::Object>’ has no member named ‘IsNearDeath’
     assert(persistent().IsNearDeath());
                         ^
In file included from ../../nan/nan.h:2698:0,
                 from ../src/addon.cc:2:
../../nan/nan_object_wrap.h: In member function ‘void Nan::ObjectWrap::MakeWeak()’:
../../nan/nan_object_wrap.h:67:34: warning: ‘void v8::PersistentBase<T>::MarkIndependent() [with T = v8::Object]’ is deprecated: Weak objects are always considered independent. Use TracedGlobal when trying to use EmbedderHeapTracer. Use a strong handle when trying to keep an object alive. [-Wdeprecated-declarations]
     persistent().MarkIndependent();
                                  ^
In file included from /home/dnl/.node-gyp/12.7.0/include/node/v8-internal.h:14:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/v8.h:25,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node.h:63,
                 from ../src/addon.cc:1:
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:567:22: note: declared here
       V8_INLINE void MarkIndependent());
                      ^
/home/dnl/.node-gyp/12.7.0/include/node/v8config.h:311:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from /usr/include/c++/7/cassert:44:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node_object_wrap.h:26,
                 from ../../nan/nan.h:54,
                 from ../src/addon.cc:2:
../../nan/nan_object_wrap.h: In static member function ‘static void Nan::ObjectWrap::WeakCallback(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)’:
../../nan/nan_object_wrap.h:124:26: error: ‘class Nan::Persistent<v8::Object>’ has no member named ‘IsNearDeath’
     assert(wrap->handle_.IsNearDeath());
                          ^
secp256k1.target.mk:158: recipe for target 'Release/obj.target/secp256k1/src/addon.o' failed
make: *** [Release/obj.target/secp256k1/src/addon.o] Error 1
make: Leaving directory '/home/dnl/workspace/test/node_modules/secp256k1/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:266:23)
gyp ERR! stack     at ChildProcess.emit (events.js:203:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Linux 4.15.0-54-generic
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/dnl/workspace/test/node_modules/secp256k1
gyp ERR! node -v v12.7.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! secp256k1@2.0.4 rebuild: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the secp256k1@2.0.4 rebuild script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/dnl/.npm/_logs/2019-08-03T21_14_51_562Z-debug.log

> eccrypto@1.1.1 install /home/dnl/workspace/test/node_modules/eccrypto
> node-gyp rebuild || exit 0

make: Entering directory '/home/dnl/workspace/test/node_modules/eccrypto/build'
  CXX(target) Release/obj.target/ecdh/ecdh.o
In file included from ../../nan/nan_converters.h:67:0,
                 from ../../nan/nan.h:221,
                 from ../ecdh.cc:2:
../../nan/nan_converters_43_inl.h: In static member function ‘static Nan::imp::ToFactoryBase<v8::Boolean>::return_t Nan::imp::ToFactory<v8::Boolean>::convert(v8::Local<v8::Value>)’:
../../nan/nan_converters_43_inl.h:18:51: warning: ‘v8::MaybeLocal<v8::Boolean> v8::Value::ToBoolean(v8::Local<v8::Context>) const’ is deprecated: ToBoolean can never throw. Use Local version. [-Wdeprecated-declarations]
       val->To ## TYPE(isolate->GetCurrentContext())                            \
                                                   ^
../../nan/nan_converters_43_inl.h:22:1: note: in expansion of macro ‘X’
 X(Boolean)
 ^
In file included from /home/dnl/.node-gyp/12.7.0/include/node/v8-internal.h:14:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/v8.h:25,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node.h:63,
                 from ../ecdh.cc:1:
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:2531:59: note: declared here
                 V8_WARN_UNUSED_RESULT MaybeLocal<Boolean> ToBoolean(
                                                           ^
/home/dnl/.node-gyp/12.7.0/include/node/v8config.h:311:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../../nan/nan_converters.h:67:0,
                 from ../../nan/nan.h:221,
                 from ../ecdh.cc:2:
../../nan/nan_converters_43_inl.h: In static member function ‘static Nan::imp::ValueFactoryBase<bool>::return_t Nan::imp::ToFactory<bool>::convert(v8::Local<v8::Value>)’:
../../nan/nan_converters_43_inl.h:37:57: warning: ‘v8::Maybe<bool> v8::Value::BooleanValue(v8::Local<v8::Context>) const’ is deprecated: BooleanValue can never throw. Use Isolate version. [-Wdeprecated-declarations]
   return val->NAME ## Value(isolate->GetCurrentContext());                     \
                                                         ^
../../nan/nan_converters_43_inl.h:40:1: note: in expansion of macro ‘X’
 X(bool, Boolean)
 ^
In file included from /home/dnl/.node-gyp/12.7.0/include/node/v8-internal.h:14:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/v8.h:25,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node.h:63,
                 from ../ecdh.cc:1:
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:2569:51: note: declared here
                 V8_WARN_UNUSED_RESULT Maybe<bool> BooleanValue(
                                                   ^
/home/dnl/.node-gyp/12.7.0/include/node/v8config.h:311:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../../nan/nan_new.h:189:0,
                 from ../../nan/nan.h:222,
                 from ../ecdh.cc:2:
../../nan/nan_implementation_12_inl.h: In static member function ‘static Nan::imp::FactoryBase<v8::Function>::return_t Nan::imp::Factory<v8::Function>::New(Nan::FunctionCallback, v8::Local<v8::Value>)’:
../../nan/nan_implementation_12_inl.h:105:32: error: no matching function for call to ‘v8::Function::New(v8::Isolate*&, void (&)(const v8::FunctionCallbackInfo<v8::Value>&), v8::Local<v8::Object>&)’
                           , obj));
                                ^
In file included from /home/dnl/.node-gyp/12.7.0/include/node/node.h:63:0,
                 from ../ecdh.cc:1:
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:4132:31: note: candidate: static v8::MaybeLocal<v8::Function> v8::Function::New(v8::Local<v8::Context>, v8::FunctionCallback, v8::Local<v8::Value>, int, v8::ConstructorBehavior, v8::SideEffectType)
   static MaybeLocal<Function> New(
                               ^~~
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:4132:31: note:   no known conversion for argument 1 from ‘v8::Isolate*’ to ‘v8::Local<v8::Context>’
In file included from ../ecdh.cc:2:0:
../../nan/nan.h: In constructor ‘Nan::Utf8String::Utf8String(v8::Local<v8::Value>)’:
../../nan/nan.h:1064:78: warning: ‘v8::Local<v8::String> v8::Value::ToString(v8::Isolate*) const’ is deprecated: Use maybe version [-Wdeprecated-declarations]
       v8::Local<v8::String> string = from->ToString(v8::Isolate::GetCurrent());
                                                                              ^
In file included from /home/dnl/.node-gyp/12.7.0/include/node/v8-internal.h:14:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/v8.h:25,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node.h:63,
                 from ../ecdh.cc:1:
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:2551:31: note: declared here
                 Local<String> ToString(Isolate* isolate) const);
                               ^
/home/dnl/.node-gyp/12.7.0/include/node/v8config.h:311:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../ecdh.cc:2:0:
../../nan/nan.h: In member function ‘void Nan::AsyncWorker::SaveToPersistent(const char*, const v8::Local<v8::Value>&)’:
../../nan/nan.h:1855:64: warning: ‘bool v8::Object::Set(v8::Local<v8::Value>, v8::Local<v8::Value>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     New(persistentHandle)->Set(New(key).ToLocalChecked(), value);
                                                                ^
In file included from /home/dnl/.node-gyp/12.7.0/include/node/v8-internal.h:14:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/v8.h:25,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node.h:63,
                 from ../ecdh.cc:1:
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:3366:26: note: declared here
                     bool Set(Local<Value> key, Local<Value> value));
                          ^
/home/dnl/.node-gyp/12.7.0/include/node/v8config.h:326:3: note: in definition of macro ‘V8_DEPRECATE_SOON’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../ecdh.cc:2:0:
../../nan/nan.h: In member function ‘void Nan::AsyncWorker::SaveToPersistent(const v8::Local<v8::String>&, const v8::Local<v8::Value>&)’:
../../nan/nan.h:1861:42: warning: ‘bool v8::Object::Set(v8::Local<v8::Value>, v8::Local<v8::Value>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     New(persistentHandle)->Set(key, value);
                                          ^
In file included from /home/dnl/.node-gyp/12.7.0/include/node/v8-internal.h:14:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/v8.h:25,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node.h:63,
                 from ../ecdh.cc:1:
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:3366:26: note: declared here
                     bool Set(Local<Value> key, Local<Value> value));
                          ^
/home/dnl/.node-gyp/12.7.0/include/node/v8config.h:326:3: note: in definition of macro ‘V8_DEPRECATE_SOON’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../ecdh.cc:2:0:
../../nan/nan.h: In member function ‘void Nan::AsyncWorker::SaveToPersistent(uint32_t, const v8::Local<v8::Value>&)’:
../../nan/nan.h:1867:44: warning: ‘bool v8::Object::Set(uint32_t, v8::Local<v8::Value>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     New(persistentHandle)->Set(index, value);
                                            ^
In file included from /home/dnl/.node-gyp/12.7.0/include/node/v8-internal.h:14:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/v8.h:25,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node.h:63,
                 from ../ecdh.cc:1:
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:3375:26: note: declared here
                     bool Set(uint32_t index, Local<Value> value));
                          ^
/home/dnl/.node-gyp/12.7.0/include/node/v8config.h:326:3: note: in definition of macro ‘V8_DEPRECATE_SOON’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../ecdh.cc:2:0:
../../nan/nan.h: In member function ‘v8::Local<v8::Value> Nan::AsyncWorker::GetFromPersistent(const char*) const’:
../../nan/nan.h:1873:61: warning: ‘v8::Local<v8::Value> v8::Object::Get(v8::Local<v8::Value>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
         New(persistentHandle)->Get(New(key).ToLocalChecked()));
                                                             ^
In file included from /home/dnl/.node-gyp/12.7.0/include/node/v8-internal.h:14:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/v8.h:25,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node.h:63,
                 from ../ecdh.cc:1:
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:3419:55: note: declared here
   V8_DEPRECATE_SOON("Use maybe version", Local<Value> Get(Local<Value> key));
                                                       ^
/home/dnl/.node-gyp/12.7.0/include/node/v8config.h:326:3: note: in definition of macro ‘V8_DEPRECATE_SOON’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../ecdh.cc:2:0:
../../nan/nan.h: In member function ‘v8::Local<v8::Value> Nan::AsyncWorker::GetFromPersistent(const v8::Local<v8::String>&) const’:
../../nan/nan.h:1879:55: warning: ‘v8::Local<v8::Value> v8::Object::Get(v8::Local<v8::Value>)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     return scope.Escape(New(persistentHandle)->Get(key));
                                                       ^
In file included from /home/dnl/.node-gyp/12.7.0/include/node/v8-internal.h:14:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/v8.h:25,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node.h:63,
                 from ../ecdh.cc:1:
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:3419:55: note: declared here
   V8_DEPRECATE_SOON("Use maybe version", Local<Value> Get(Local<Value> key));
                                                       ^
/home/dnl/.node-gyp/12.7.0/include/node/v8config.h:326:3: note: in definition of macro ‘V8_DEPRECATE_SOON’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from ../ecdh.cc:2:0:
../../nan/nan.h: In member function ‘v8::Local<v8::Value> Nan::AsyncWorker::GetFromPersistent(uint32_t) const’:
../../nan/nan.h:1884:57: warning: ‘v8::Local<v8::Value> v8::Object::Get(uint32_t)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     return scope.Escape(New(persistentHandle)->Get(index));
                                                         ^
In file included from /home/dnl/.node-gyp/12.7.0/include/node/v8-internal.h:14:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/v8.h:25,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node.h:63,
                 from ../ecdh.cc:1:
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:3423:55: note: declared here
   V8_DEPRECATE_SOON("Use maybe version", Local<Value> Get(uint32_t index));
                                                       ^
/home/dnl/.node-gyp/12.7.0/include/node/v8config.h:326:3: note: in definition of macro ‘V8_DEPRECATE_SOON’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from /usr/include/c++/7/cassert:44:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node_object_wrap.h:26,
                 from ../../nan/nan.h:54,
                 from ../ecdh.cc:2:
../../nan/nan_object_wrap.h: In destructor ‘virtual Nan::ObjectWrap::~ObjectWrap()’:
../../nan/nan_object_wrap.h:24:25: error: ‘class Nan::Persistent<v8::Object>’ has no member named ‘IsNearDeath’
     assert(persistent().IsNearDeath());
                         ^
In file included from ../../nan/nan.h:2698:0,
                 from ../ecdh.cc:2:
../../nan/nan_object_wrap.h: In member function ‘void Nan::ObjectWrap::MakeWeak()’:
../../nan/nan_object_wrap.h:67:34: warning: ‘void v8::PersistentBase<T>::MarkIndependent() [with T = v8::Object]’ is deprecated: Weak objects are always considered independent. Use TracedGlobal when trying to use EmbedderHeapTracer. Use a strong handle when trying to keep an object alive. [-Wdeprecated-declarations]
     persistent().MarkIndependent();
                                  ^
In file included from /home/dnl/.node-gyp/12.7.0/include/node/v8-internal.h:14:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/v8.h:25,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node.h:63,
                 from ../ecdh.cc:1:
/home/dnl/.node-gyp/12.7.0/include/node/v8.h:567:22: note: declared here
       V8_INLINE void MarkIndependent());
                      ^
/home/dnl/.node-gyp/12.7.0/include/node/v8config.h:311:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from /usr/include/c++/7/cassert:44:0,
                 from /home/dnl/.node-gyp/12.7.0/include/node/node_object_wrap.h:26,
                 from ../../nan/nan.h:54,
                 from ../ecdh.cc:2:
../../nan/nan_object_wrap.h: In static member function ‘static void Nan::ObjectWrap::WeakCallback(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)’:
../../nan/nan_object_wrap.h:124:26: error: ‘class Nan::Persistent<v8::Object>’ has no member named ‘IsNearDeath’
     assert(wrap->handle_.IsNearDeath());
                          ^
../ecdh.cc: At global scope:
../ecdh.cc:6:11: error: ‘v8::Handle’ has not been declared
 using v8::Handle;
           ^~~~~~
ecdh.target.mk:113: recipe for target 'Release/obj.target/ecdh/ecdh.o' failed
make: *** [Release/obj.target/ecdh/ecdh.o] Error 1
make: Leaving directory '/home/dnl/workspace/test/node_modules/eccrypto/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:266:23)
gyp ERR! stack     at ChildProcess.emit (events.js:203:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Linux 4.15.0-54-generic
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/dnl/workspace/test/node_modules/eccrypto
gyp ERR! node -v v12.7.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN test@1.0.0 No description
npm WARN test@1.0.0 No repository field.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: secp256k1@2.0.4 (node_modules/secp256k1):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: secp256k1@2.0.4 install: `npm run rebuild`
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Exit status 1

+ eccrypto@1.1.1

```

**Comment by JBaczuk** [comment]
Confirmed, thanks for reporting

**Comment by pubkey** [comment]
@JBaczuk any update on this?
Do you think this can be solved here or does this have to be solved at secp256k1?

**Comment by JBaczuk** [comment]
@pubkey It appears it was [fixed](https://github.com/cryptocoinjs/secp256k1-node/issues/142#issuecomment-490365020) in `secp256k1` I'm trying to upgrade to their latest, and there are some other things broken, I'll keep you posted.

**Comment by JBaczuk** [comment]
phew ok, I got it working by updating the `secp256k1` also managed to updated the automated tests at the same time. The fix is in the `v1.1.2` release.
